### PR TITLE
Basic Block Breaking and Placing (I don't know how to lint a file)

### PR DESCRIPTION
**Pull request by ShrimpyStuff**

### CODE REVIEW

**Comment by u9g** [comment]
I PR'd linting https://github.com/ShrimpyStuff/prismarine-viewer/pull/1

**Comment by ShrimpyStuff** [comment]
Thx

File: examples/web_client/index.js [hunk 1]
```diff
@@ -121,6 +122,26 @@ async function main () {
       }
     }, false)
 
+    document.addEventListener('mousedown', (e) => {
+      const BlockDistance = (bot.gamemode == 1) ? 8 : 5
```
Comment (extremeheat) [inline examples/web_client/index.js:126]: where do these constants come from?

File: examples/web_client/index.js [hunk 2]
```diff
@@ -121,6 +122,26 @@ async function main () {
...
+      const ButtonBlock = bot.blockAtCursor(BlockDistance)
+      if (!ButtonBlock) return
+      if (e.button === 0) {
+        if (bot.canDigBlock(ButtonBlock)) {
+          bot.dig(ButtonBlock)
```
Comment (extremeheat) [inline examples/web_client/index.js:131]: what happens when the user moves the cursor away from the block?

File: examples/web_client/index.js [hunk 3]
```diff
@@ -121,6 +122,26 @@ async function main () {
...
+        }
+      } else if (e.button === 2) {
+        const vecArray = [new Vec3(0, -1, 0), new Vec3(0, 1, 0), new Vec3(0, 0, -1), new Vec3(0, 0, 1), new Vec3(-1, 0, 0), new Vec3(1, 0, 0)]
```
Comment (extremeheat) [inline examples/web_client/index.js:134]: I think this constant should be defined at the top or elsewhere for reusability

**Review by extremeheat — CHANGES_REQUESTED** [review]

File: examples/web_client/index.js (hunk 1 above)
Comment (Karang) [reply examples/web_client/index.js:126]: Just checked the wiki https://minecraft.gamepedia.com/Breaking#Basics_of_breaking it should be 5 for creative and 4.5 for survival.

Comment (ShrimpyStuff) [reply examples/web_client/index.js:126]: Oh someone on the server told me 8 for creative and 5 for survival

File: examples/web_client/index.js (hunk 2 above)
Comment (ShrimpyStuff) [reply examples/web_client/index.js:131]: I have no idea

File: examples/web_client/index.js (hunk 1 above)
Comment (Karang) [reply examples/web_client/index.js]: Yeah, that was me, but you should have checked :)

**Comment by Karang** [comment]
A more general comment: I have nothing against implementing more client functionalities into this example. But it is just an example, it doesn't aim to be a complete client, just to show a start for people to implement their client using prismarine-viewer.

I feel like this kind of functionalities should be implemented in a real client project, like https://github.com/Heath123/mineflayer-web-test

@Heath123 @rom1504 what do you think ?

**Comment by ShrimpyStuff** [comment]
I don't own this repo so I don't care what you do with this PR and I'm fine with you guys not implementing it here

**Comment by rom1504** [comment]
I agree but also https://github.com/Heath123/mineflayer-web-test seems dead.
Should we have a PrismarineJS repo dedicated to exposing features to build a fully fledged client ? (Where this kind of PR would belong)

**Comment by Heath123** [comment]
> I agree but also https://github.com/Heath123/mineflayer-web-test seems dead.
> Should we have a PrismarineJS repo dedicated to exposing features to build a fully fledged client ? (Where this kind of PR would belong)

Well I was planning to work on it and merge it into main Mineweb but having as an official Prismarine project seems to make Psome sense too

**Comment by AwesomestCode** [comment]
I plan on continuing to work on MineWeb, I've just been very busy the past couple of weeks.

**Review by AwesomestCode — CHANGES_REQUESTED** [review]
tested; doesn’t work with spawn protection: it gets broken on pviewer side but server side it doesn’t work, leaving a glitched block.

**Comment by ShrimpyStuff** [comment]
I have no idea how to fix that

**Comment by ShrimpyStuff** [comment]
Moved to https://github.com/PrismarineJS/prismarine-web-client

**Pull request closed without merging**